phoenicis-library/src/main/java/org/phoenicis/library/LibraryManager.java

```diff
@@ -108,6 +109,12 @@ private ShortcutDTO fetchShortcutDTO(File shortcutDirectory, File file) {
             final URI miniature = miniatureFile.exists() ? miniatureFile.toURI()
                     : getClass().getResource("defaultMiniature.png").toURI();
 
+            String name = baseName;
+            if (nameFile.exists()) {
+                name = IOUtils.toString(new FileInputStream(nameFile), "UTF-8");
+                name = name.replace("\n", "");
```

Comment:
Are there other "bad" characters?

Reply:
I think you misunderstood this. The "\n" is just removed because you might have a line break in the `shortcut.name`. It has nothing to do with "bad" characters. Those are filtered in the `ShortcutDTO`.